cowprotocol/cowswap
# fix(frontend): detect a connected user in a safe

# Summary
If the user navigates to the CoW app in safe wallet and the user is not connected to the Safe, on connection the CoW site doesn’t detect this and ask the user to refresh the page. By polling the safeInfo we can detect that the connected safe changed state.

![grafik](https://github.com/cowprotocol/cowswap/assets/693770/8e08910e-5759-4ae5-8394-29c49a14ea90)

It actually should no longer be necessary to refresh.

# To Test
1. Navigate to a safe (where not connected)
2. Connect when on the widget/app page
3. After few sec. CoW should change the "connect signer" text to "select token"
